cli/crud/import/import.go

```diff
@@ -0,0 +1,1214 @@
+	batchCounter uint32
+	tupleNumber  uint32
+	header       []string
+}
+
+// getCurrentTupleIndex allows to get the index of the current tuple within the batch.
+func (ctx *t_batchSequenceCtx) getCurrentTupleIndex() uint32 {
+	return (ctx.tupleNumber - 1) % ctx.batchSize
+}
+
+// t_unparsedCsvReaderCtx contains private fields for organizing the context of raw reading of a file with data for import.
+// Raw reading is necessary to get a string of data in an unparsed form. The standard parser does not allow this.
+// It also contains fields for implementing a mechanism to eliminate the inconsistency of the position counter due to a multi-line CSV records.
+// Fields inside this structure should not be read or overwritten by the programmer (exclusion are methods updCtxAfterParseError, updCtxAfterRecordHandling).
+type t_unparsedCsvReaderCtx struct {
+	_masterPosition uint32
+	_slavePosition  uint32
+	_currentRecord  string
+	_scanner        *bufio.Scanner
+}
+
+// updCtxAfterParseError updates the raw read context in case of a record reading error of the main parser.
+// WARNING: this method must not be called multiple times in one iteration of the main csv parser!
+func (ctx *t_unparsedCsvReaderCtx) updCtxAfterParseError() (uint32, string) {
+	ctx._masterPosition++
+	ctx._scanner.Scan()
+	ctx._currentRecord += ctx._scanner.Text()
+	currentPosition, currentRec := ctx._masterPosition, ctx._currentRecord
+	ctx._currentRecord = ""
+
+	return currentPosition, currentRec
+}
+
+// updCtxAfterRecordHandling updates the raw read context in case of a record success reading of the main parser.
+// WARNING: this method must not be called multiple times in one iteration of the main csv parser!
+func (ctx *t_unparsedCsvReaderCtx) updCtxAfterRecordHandling(csvReader *ttcsv.Reader) (uint32, string) {
+	ctx._masterPosition++
+	ctx._scanner.Scan()
+	ctx._currentRecord += ctx._scanner.Text()
+
+	ctx._slavePosition = ctx._masterPosition
+	for uint32(csvReader.NumLine) > ctx._masterPosition {
+		// value of current position in raw reader should catch up with the reading position of the main parser
+		// if a single entry in a CSV file could not make up several lines, then this mechanism would not be needed
+		// example: 123,"da\nta",321 is one csv record, but unparsed form is two lines, parsed form is slice ["123", "da\nta", 321]
+		ctx._masterPosition++
+		ctx._scanner.Scan()
+		ctx._currentRecord += "\n" + ctx._scanner.Text()
+	}
+
+	var currentPosition uint32
+	if ctx._masterPosition != ctx._slavePosition {
+		currentPosition = ctx._slavePosition
+	} else {
+		currentPosition = ctx._masterPosition
+	}
+	currentRec := ctx._currentRecord
+	ctx._currentRecord = ""
+
+	return currentPosition, currentRec
+}
+
+var importSummary t_importSummary = t_importSummary{}
+
+// RunImport implements data import via crud.
+func RunImport(cmdCtx *cmdcontext.CmdCtx, conn *connector.Conn, csvReader *ttcsv.Reader, rawReaderFd *os.File, crudImportFlags *Opts, spaceName string) error {
```

Comment:
The function ~ 600 lines it's not cool. MOve some part of logic to  a separated functions.

Reply:
At the moment, the largest function `mainParsingCycle` is ~440 rows. I don't find an opportunity to improve the result because of the operators `continue` and `break` in this func